crates/project/src/project.rs

```diff
@@ -3201,6 +3201,59 @@ impl Project {
         let disk_based_diagnostics_progress_token =
             adapter.disk_based_diagnostics_progress_token.clone();
 
+        // Log LSP messages under experimental/serverStatus and display pop-up for errors
+        language_server
+            .on_notification::<ServerStatus, _>({
+                let this = this.clone();
+                let name = name.to_string();
+                move |params, mut cx| {
+                    let this = this.clone();
+                    let name = name.to_string();
+                    match params.health.as_str() {
+                        "ok" => {
+                            log::info!("Health: {}", params.health);
+                        }
+                        "warning" => {
+                            log::warn!(
+                                "Health: {}.\nMessage: {}",
+                                params.health,
+                                params.message.unwrap_or_default()
+                            );
+                        }
+                        "error" => {
+                            log::error!(
+                                "Health: {}.\nMessage: {}",
+                                params.health,
+                                params.message.clone().unwrap_or_default()
+                            );
+                            let (tx, _rx) = smol::channel::bounded(1);
+                            let request = LanguageServerPromptRequest {
+                                level: match params.health.as_str() {
+                                    "error" => PromptLevel::Critical,
+                                    "warning" => PromptLevel::Warning,
+                                    "ok" => PromptLevel::Info,
+                                    _ => PromptLevel::Info,
+                                },
+                                message: params.message.unwrap_or_default(),
+                                actions: Vec::new(),
```

Comment:
Not sure if passing an empty vec is the right thing to do

Reply:
Could you check for `actions` usage and understand the code around it?
By the looks of it, it seems that `actions` are used as a way to respond to the LSP server with something, which is not our case?

Reply:
Yupp! I did look around and it seems like there is nothing for us to respond with(: